projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.ts

```diff
@@ -134,11 +136,20 @@ export class CommentComponent implements OnChanges {
 
   formNgStyle() {
     return {
-      top: (this.rectTop * this.zoom) + 'px',
+      top: this.getTotalPreviousPagesHeight() + (this.rectTop * this.zoom) + 'px',
       'z-index': this.selected ? 100 : 0
     };
   }
 
+  getTotalPreviousPagesHeight(): number {
+    const pageMarginBottom = 10;
+    let totalPreviousPagesHeight = 0;
+    for (let i = 0; i < this.page - 1; i++) {
+      totalPreviousPagesHeight += this.pageHeights[i] + pageMarginBottom;
+    }
+    return totalPreviousPagesHeight;
```

Comment:
let's go functional!

```
return this.pageHeights.reduce((total, height) => total + height + pageMarginBottom, 0);
```

Reply:
just curious, can you stop at a certain index in the array with reduce? if not would it still be better to use reduce but have the function contain an condition?

Reply:
the reduce function gives three args (total, height, index) so you could use the index, but I'd say it would be better to do:

```
return this.pageHeights
  .slice(0, -1)
  .reduce((total, height) => total + height + pageMarginBottom, 0);
```